lib/chess_web/controllers/game_controller.ex

```diff
@@ -26,12 +40,20 @@ defmodule ChessWeb.GameController do
         |> put_flash(:info, "Game created successfully.")
         |> redirect(to: game_path(conn, :show, game))
       {:error, changeset} ->
-        render(conn, "new.html", changeset: changeset)
+        opponents = get_opponents(current_user(conn))
```

Comment:
What do you think of

```
opponents = 
  conn
  |> current_user()
  |> get_opponents()

conn
|> render(:new, changeset: changeset, opponents: opponents)
```

You could also extract something like this:

```
defp render_new(conn, changeset, opponents) do
  conn
  |> render(:new, changeset: changeset, opponents: opponents)
end
```
and use this in the new action and the failed create actions. You could do something similar for `render_edit` and use it for the failed update action and the edit action in different controllers.